mk/aosp_optee.mk

```diff
@@ -93,7 +93,7 @@ $(TA_TMP_FILE): BUILD_OPTEE_OS
 	@echo "Start building TA for $(PRIVATE_TA_SRC_DIR) $(PRIVATE_TA_TMP_FILE)..."
 	$(MAKE) -C $(TOP_ROOT_ABS)/$(PRIVATE_TA_SRC_DIR) O=$(TOP_ROOT_ABS)/$(OPTEE_TA_OUT_DIR)/$(PRIVATE_TA_TMP_DIR) \
 		TA_DEV_KIT_DIR=$(TOP_ROOT_ABS)/$(TA_DEV_KIT_DIR) \
-		$(CROSS_COMPILE_LINE)
+		$(CROSS_COMPILE_LINE) NOWERROR=1
```

Comment:
Yes. This line is used to build all the TAs in xtest and now the sks TA as well. There are still many warnings in the sks TA (see below) which I don't know to fix. Doesn't your compiler throw these errors too? The aosp compiler is 4.9.

```
src/entry.c: In function 'TA_OpenSessionEntryPoint':
src/entry.c:45:13: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
  *session = (void *)handle_get(&sks_session_db, sess);
             ^
src/entry.c: In function 'TA_CloseSessionEntryPoint':
src/entry.c:52:29: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
  ck_token_close_tee_session((int)session);
                             ^
src/entry.c:53:39: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
  TEE_Free(handle_put(&sks_session_db, (int)session));
                                       ^
src/entry.c: In function 'TA_InvokeCommandEntryPoint':
src/entry.c:75:16: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
  int teesess = (int)session;
                ^
src/processing_aes.c: In function 'tee_ae_decrypt_final':
src/processing_aes.c:337:36: warning: passing argument 3 of 'reveale_ae_data' from incompatible pointer type
   return reveale_ae_data(ctx, out, out_size);
                                    ^
src/processing_aes.c:284:17: note: expected 'size_t *' but argument is of type 'uint32_t *'
 static uint32_t reveale_ae_data(struct ae_aes_context *ctx,
                 ^
src/processing_aes.c:398:33: warning: passing argument 3 of 'reveale_ae_data' from incompatible pointer type
  rv = reveale_ae_data(ctx, out, out_size);
                                 ^
src/processing_aes.c:284:17: note: expected 'size_t *' but argument is of type 'uint32_t *'
 static uint32_t reveale_ae_data(struct ae_aes_context *ctx,
                 ^
```

Reply:
This change affects the generic optee_os, not specifically the sks and I would rather get rid of those warning from the sks source or get this change be merged in the optee_os mainline.

Could you create a specific p-r for this change so that I can merge your other commits?
We will try to find the best way to tackle these warnings.

Reply:
Sorry but I don't quite understand.

> This change affects the generic optee_os, not specifically the sks and I would rather get rid of those warning from the sks source or get this change be merged in the optee_os mainline.

This change only affect compilation of TAs in aosp build env, not optee_os, but I agree it's better to get rid of warnings.

> Could you create a specific p-r for this change so that I can merge your other commits?

Ok. I'll remove this change from this p-r, but about creating a specific p-r for this change, where do you want me to submit it to? optee_os mainline?